### DISCUSSION TITLE
Feature Request: ability to add custom settings

### DISCUSSION THREAD

**Discussion by mgrider** [kean/Pulse#85] [Ideas]
I'm using Pulse and PulseUI in a current project and have been very happy with it! I have a "hidden" gesture that opens PulseUI, and it's proven very handy already, and I imagine will prove even more so when the app is in the hands of beta testers down the road.

I've used similar debug gestures in previous apps, and in addition to displaying logs, it's been common for those debug windows to give the user access to hidden settings or feature flags. I'm wondering if custom settings is a feature you've considered adding to Pulse/PulseUI?

I could imagine initializing `MainViewController` with my own customized/injected version of `MainViewModel` or `SettingsViewModel` with some new array of custom items, but of course it could look a lot of different ways.

**Comment by kean** [comment]
Hi, in addition to MainView, there are public APIs for individual tabs as well. You can compose them in any way you'd like. That's actually how I personally use it, because I also typically need a view for debug settings and for feature flags.
